src/context/InitalLoadingContext.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { createContext, useContext, useState } from 'react'
+
+const initialContext = {
+  initialLoadingComplete: false,
+  markInitialLoadingComplete: () => {
+    // set in context provider construction
+  },
+}
+
+const InitialLoadingContext = createContext(initialContext)
+
+export const useInitialLoading = () => useContext(InitialLoadingContext)
+
+export const InitialLoadingProvider = ({ children }) => {
+  const [initialLoadingComplete, setInitialLoadingComplete] = useState(false)
+
+  const markInitialLoadingComplete = () => {
+    setInitialLoadingComplete(true)
+  }
+
+  return (
+    <InitialLoadingContext.Provider
+      value={{ initialLoadingComplete, markInitialLoadingComplete }}
```

Comment:
prefer memoizing this object, otherwise each render of this component will create a new object, which will cause a rerender of all users of this context, which is unnecessary